.github/workflows/changelog-check.yml

```diff
@@ -0,0 +1,68 @@
+# **what?**
+# Checks that a file has been committed under the /.changes directory
+# as a new CHANGELOG entry.  Cannot check for a specific filename as
+# it is dynamically generated by change type and timestamp.
+# This workflow should not require any secrets since it runs for PRs
+# from forked repos.
+# By default, secrets are not passed to workflows running from
+# a forked repo.
+
+# **why?**
+# Ensure code change gets reflected in the CHANGELOG.
+
+# **when?**
+# This will run for all PRs going into main and *.latest.
+
+name: Check Changelog Entry
+
+on:
+  push:
```

Comment:
Do we need this to only run when checking PRs? Not sure running manually or on commits to `main` are helpful